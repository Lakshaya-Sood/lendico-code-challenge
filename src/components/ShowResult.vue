<template>
  <b-card align="center">
    <b-spinner v-if="isLoading" style="width: 3rem; height: 3rem;"></b-spinner>
    <div v-else>
      <h2 v-if="!!result">
        {{this.description}}
        <span style="color:black;">{{this.result}}</span>
      </h2>
      <h3 v-else>{{this.failMessage}}</h3>
    </div>
  </b-card>
</template>

<script lang='ts'>
import Vue from "vue";
import { Prop, Component } from "vue-property-decorator";

@Component({
  name: "show-result"
})
export default class ShowResult extends Vue {
  @Prop({ required: true }) readonly description!: string;
  @Prop({ required: true }) readonly failMessage!: string;
  @Prop({ required: true }) readonly result!: number | null;
  @Prop({ required: true }) readonly isLoading!: boolean;
}
</script>

<style>
</style>
