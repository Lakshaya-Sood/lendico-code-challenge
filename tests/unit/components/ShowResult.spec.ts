import { shallowMount, createLocalVue } from '@vue/test-utils'
import BootstrapVue from 'bootstrap-vue'

import ShowResult from '@/components/ShowResult.vue'

const vue_mock = createLocalVue()
vue_mock.use(BootstrapVue)


describe('show-result component', () => {
    const wrapper = shallowMount(ShowResult, {
        propsData: {
            description: 'Some Discription',
            failMessage: "Some Error Message",
            result: 3456.12,
            isLoading: false,
        },
        localVue: vue_mock
    })
    test('name in kebab case', () => {
        expect(wrapper.vm.$options.name).toEqual('show-result')
    })

    test('no changes as compared to last snapshot', () => {
        expect(wrapper.element).toMatchSnapshot() // https://vue-test-utils.vuejs.org/guides/#snapshot-testing
    })
})
