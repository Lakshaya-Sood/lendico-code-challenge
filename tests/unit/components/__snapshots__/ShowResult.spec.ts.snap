// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`show-result component no changes as compared to last snapshot 1`] = `
<b-card-stub
  align="center"
  bodytag="div"
  footertag="div"
  headertag="div"
  subtitle=""
  subtitletag="h6"
  subtitletextvariant="muted"
  tag="div"
  title=""
  titletag="h4"
>
  <div>
    <h2>
      
      Some Discription
      
      <span
        style="color: black;"
      >
        3456.12
      </span>
    </h2>
  </div>
</b-card-stub>
`;
